HouseRules_Essentials/Rules/PartyElectricityDamageOverriddenRule.cs

```diff
@@ -0,0 +1,104 @@
+    {
+        public override string Description => "Player on player electricity damage is zero.";
+
+        private static bool _globalAdjustments;
+        private static bool _isActivated;
+
+        private readonly bool _adjustments;
+        private static GameContext _gameContext;
+
+        public PartyElectricityDamageOverriddenRule(bool adjustments)
+        {
+            _adjustments = adjustments;
+        }
+
+        public bool GetConfigObject() => _adjustments;
+
+        protected override void OnActivate(GameContext gameContext)
+        {
+            _globalAdjustments = _adjustments;
+            _isActivated = true;
+            _gameContext = gameContext;
+        }
+
+        protected override void OnDeactivate(GameContext gameContext) => _isActivated = false;
+
+        private static void Patch(Harmony harmony)
+        {
+
+            harmony.Patch(
+                original: AccessTools.Method(typeof(ProjectileHitSequence), "OnStarted"),
+                prefix: new HarmonyMethod(
+                    typeof(PartyElectricityDamageOverriddenRule),
+                    nameof(ProjectileHitSequence_OnStarted_Prefix)));
+
+            harmony.Patch(
+                original: AccessTools.Method(typeof(SerializableEventQueue), "ValidateSerializableEvent"),
+                prefix: new HarmonyMethod(
+                    typeof(PartyElectricityDamageOverriddenRule),
+                    nameof(SerializableEventQueue_ValidateSerializableEvent_Postfix)));
+        }
+
+        private static bool ProjectileHitSequence_OnStarted_Prefix(ref ProjectileHitSequence __instance)
+        {
+            if (!_isActivated)
+            {
+                return true;
+            }
+
+            var targetPiece = Traverse.Create(__instance).Field("targetPiece").GetValue<Piece>();
+            var abilityContext = Traverse.Create(__instance).Field("abilityContext").GetValue<AbilityContext>();
+            var damage = Traverse.Create(abilityContext).Field("damage").GetValue<Damage>();
+            var attackerTarget = Traverse.Create(abilityContext).Field("attacker").GetValue<Target>();
+            var attacker = Traverse.Create(attackerTarget).Field("piece").GetValue<Piece>();
+
+            if (attacker.IsPlayer() && targetPiece.IsPlayer() && damage.HasTag(DamageTag.Electricity))
+            {
+                var localizedText = Traverse.Create(damage).Method("GetLocalizedText", paramTypes: new[] { typeof(string), typeof(bool) }, arguments: new object[] { "Ui/pieceUi/notification/damage/noDamage", false }).GetValue<string>();
+                Notification.ShowGoldenText(new Target(targetPiece).gameObject, localizedText);
+                _gameContext.serializableEventQueue.SendEventRequest(new SerializeableEventSetStatusEffects(targetPiece.networkID, new EffectStateType[] { }, new EffectStateType[] { EffectStateType.Stunned }));
+
+                return false; // Don't run the original OnStarted method.
```

Comment:
When trying out this rule, my client was seeing damage while my host wasn't.

After some quick digging, I believe it due to all players invoking `ProjectileHitSequence#OnStarted`.  If the host patches/skips that method, it doesn't stop clients from doing so and applying the damage (exactly like the status effect that we're following up on and removing).

If we update a player's health, even by zero, the game will internally raise the `needsSync` flag for that particular stat (health).  This stat-auto-syncing is the reason why we don't have to force a board sync when hosts compute different damages.  Several ways to touch the stat, this is just the first I found: `targetPiece.effectSink.SubtractHealth(0);` 

(If only this kind of auto-syncing existed for everything else in the game, hehe).

Reply:
Oh nice.. I've added that into the rule now and tested. Works a charm.